extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -302,37 +470,101 @@ export async function newDeployment(
     if (!step) {
-      return undefined;
+      // if we have not covered the step, then don't number it.
+      return {
+        step: 0,
+        totalSteps: 0,
+      };
     }
     if (newCredentialForced(multiStepState)) {
-      if (hasMultipleEntryPoints()) {
-        return step.noCredentials.multipleEntryPoints;
+      if (hasMultiplePossibleEntryPointFiles()) {
+        if (hasMultipleContentTypesForSelectedEntryPoint()) {
+          return step.noCredentials.multipleEntryPoints.multipleContentTypes;
+        }
+        return step.noCredentials.multipleEntryPoints.singleContentType;
       }
       return step.noCredentials.singleEntryPoint;
     }
     if (newCredentialSelected(multiStepState)) {
-      if (hasMultipleEntryPoints()) {
-        return step.newCredentials.multipleEntryPoints;
+      if (hasMultiplePossibleEntryPointFiles()) {
+        if (hasMultipleContentTypesForSelectedEntryPoint()) {
+          return step.newCredentials.multipleEntryPoints.multipleContentTypes;
+        }
+        return step.newCredentials.multipleEntryPoints.singleContentType;
       }
       return step.newCredentials.singleEntryPoint;
     }
     // else it has to be existing credential selected
-    if (hasMultipleEntryPoints()) {
-      return step.existingCredentials.multipleEntryPoints;
+    if (hasMultiplePossibleEntryPointFiles()) {
+      if (hasMultipleContentTypesForSelectedEntryPoint()) {
+        return step.existingCredentials.multipleEntryPoints
+          .multipleContentTypes;
+      }
+      return step.existingCredentials.multipleEntryPoints.singleContentType;
     }
     return step.existingCredentials.singleEntryPoint;
   };
 
+  const getConfigurationInspectionQuickPicks = (
+    relEntryPoint: EntryPointPath,
+  ) => {
+    return new Promise<QuickPickItemWithIndex[]>(async (resolve, reject) => {
+      const inspectionListItems: QuickPickItemWithIndex[] = [];
+
+      try {
+        const python = await getPythonInterpreterPath();
+        const dir = path.dirname(relEntryPoint);
+        const entryPointFile = path.basename(relEntryPoint);
+
+        const inspectResponse = await api.configurations.inspect(dir, python, {
+          entrypoint: entryPointFile,
+        });
+
+        inspectionResults = inspectResponse.data;
+        inspectionResults.forEach((result, i) => {
+          const config = result.configuration;
+          if (config.entrypoint) {
+            inspectionListItems.push({
+              iconPath: new ThemeIcon("gear"),
+              label: config.type.toString(),
+              description: `(${contentTypeStrings[config.type]})`,
+              index: i,
+            });
+          }
+        });
+      } catch (error: unknown) {
+        const summary = getSummaryStringFromError(
+          "newDeployment, configurations.inspect",
+          error,
+        );
+        window.showErrorMessage(
+          `Unable to continue with project inspection failure for ${entryPoint}. ${summary}`,
```

Comment:
I get this error every time I attempt to use the entrypoint button without a Deployment already created.

![CleanShot 2024-07-31 at 16 46 46@2x](https://github.com/user-attachments/assets/120ecc37-9563-4a8e-8b4f-00c1d1e40c64)

Reply:
Looks like this was also resolved with the fix in commit: https://github.com/posit-dev/publisher/commit/40b9a0ad82d8887725ca07cc9e701bc3a1946179

Reply:
I'm getting a strange issue now where anytime I click the entrypoint button the homeview gets focus, but nothing else happens. It looks like the `newDeployment` function is bailed out of?